app/notify_admin.go

```diff
@@ -116,51 +118,90 @@ func (a *App) SendNotifyAdminPosts(c *request.Context, workspaceName string, cur
 
 	userBasedData := a.groupNotifyAdminByUser(data)
 	featureBasedData := a.groupNotifyAdminByFeature(data)
-	props := make(model.StringInterface)
+
+	pluginBasedWorkTemplateData := a.groupNotifyAdminByPlugin(data)
 
 	for _, admin := range sysadmins {
-		T := i18n.GetUserTranslations(admin.Locale)
-		message := T("app.cloud.upgrade_plan_bot_message", map[string]interface{}{"UsersNum": len(userBasedData), "WorkspaceName": workspaceName})
-		if len(userBasedData) == 1 {
-			message = T("app.cloud.upgrade_plan_bot_message_single", map[string]interface{}{"UsersNum": len(userBasedData), "WorkspaceName": workspaceName}) // todo (allan): investigate if translations library can do this
-		}
-		if trial {
-			message = T("app.cloud.trial_plan_bot_message", map[string]interface{}{"UsersNum": len(userBasedData), "WorkspaceName": workspaceName})
-			if len(userBasedData) == 1 {
-				message = T("app.cloud.trial_plan_bot_message_single", map[string]interface{}{"UsersNum": len(userBasedData), "WorkspaceName": workspaceName})
-			}
+		if len(userBasedData) > 0 && len(featureBasedData) > 0 {
+			a.upgradePlanAdminNotifyPost(c, workspaceName, userBasedData, featureBasedData, systemBot, admin, trial)
 		}
 
-		channel, appErr := a.GetOrCreateDirectChannel(c, systemBot.UserId, admin.Id)
-		if appErr != nil {
-			mlog.Warn("Error getting direct channel", mlog.Err(appErr))
-			continue
+		if len(pluginBasedWorkTemplateData) > 0 {
+			a.pluginInstallAdminNotifyPost(c, userBasedData, pluginBasedWorkTemplateData, systemBot, admin)
 		}
+	}
 
-		post := &model.Post{
-			Message:   message,
-			UserId:    systemBot.UserId,
-			ChannelId: channel.Id,
-			Type:      fmt.Sprintf("%sup_notification", model.PostCustomTypePrefix), // webapp will have to create renderer for this custom post type
+	a.FinishSendAdminNotifyPost(trial, now, pluginBasedWorkTemplateData)
+	return nil
+}
 
-		}
 
-		props["requested_features"] = featureBasedData
-		props["trial"] = trial
-		post.SetProps(props)
+func (a *App) pluginInstallAdminNotifyPost (c *request.Context, userBasedData  map[string][]*model.NotifyAdminData, pluginBasedWorkTemplateData  map[string][]*model.NotifyAdminData,  systemBot *model.Bot, admin *model.User) {
+	props := make(model.StringInterface)
 
-		_, appErr = a.CreatePost(c, post, channel, false, true)
-		if appErr != nil {
-			mlog.Warn("Error creating post", mlog.Err(appErr))
-			continue
+	channel, appErr := a.GetOrCreateDirectChannel(c, systemBot.UserId, admin.Id)
+	if appErr != nil {
+		mlog.Warn("Error getting direct channel", mlog.Err(appErr))
+		return
+	}
+
+	post := &model.Post{
+		Message:   "",
```

Comment:
Does this still need to be filled in with a message of any sort? Or is everything all configured client side?